src/GraphNavigator/SerializationGraphNavigator.php

```diff
@@ -257,8 +257,15 @@ public function accept($data, ?array $type = null)
                         continue;
                     }
 
+                    /** Metadata changes based on context, should not be cached  */
+                    $contextSpecificMetadata = $propertyMetadata;
+                    if (null !== $this->context->getPropertyNamingStrategy()) {
+                        $contextSpecificMetadata = clone $propertyMetadata;
+                        $contextSpecificMetadata->serializedName = $this->context->getPropertyNamingStrategy()->translateName($propertyMetadata);
```

Comment:
I'm a bit worried about the performance of this operation - could you add PHPBench test that will allow us to compare performance, please? See `JsonMaxDepthSerializationBench` for reference.

Reply:
Ok, added :)